### PR TITLE
[EOC-439] Profile view. Password changed successfully message is displayed after putting wrong current password filed. 

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: server/controllers/userAuth.js [hunk 1]
```diff
@@ -411,6 +416,8 @@ const changePassword = (req, res) => {
 
         return doc.save();
       }
+
+      throw new Error();
```
Comment (Ciunkos) [inline server/controllers/userAuth.js]: Please send validation exception in this case - consider inverting if so the check if for `!validHash` and throws an error from inside.

File: client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx [hunk 2]
```diff
@@ -158,24 +161,27 @@ class PasswordChangeForm extends PureComponent {
...
+                  : '',
+                newPasswordError: isNewPasswordError
+                  ? 'user.auth.input.password.invalid'
+                  : ''
+              };
+            } else {
+              newState.changePasswordErrorId =
+                'common.auth.input.wrong-password';
+            }
+          } else
```
Comment (Ciunkos) [inline client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx]: Please add braces.

File: server/controllers/userAuth.js [hunk 3]
```diff
@@ -403,17 +409,26 @@ const changePassword = (req, res) => {
...
+
+      return doc.save();
     })
     .then(() => res.send())
-    .catch(() => res.sendStatus(400));
+    .catch(err => {
+      if (err instanceof ValidationException) {
+        const { status } = err;
+
+        res.status(status).send({ reason: BadRequestReason.VALIDATION });
```
Comment (Ciunkos) [inline server/controllers/userAuth.js]: Please `return` so the next call is not made.

File: client/src/locales/en.json [hunk 4]
```diff
@@ -100,6 +100,7 @@
   "user.auth.input.password.confirm": "Confirm password",
   "user.auth.input.new-password.confirm": "Confirm new password",
   "user.auth.input.password.invalid": "Please use min 4 and max 32 characters without spaces.",
+  "common.auth.input.wrong-password": "Please enter correct current password.",
```
Comment (adamklepacz) [inline client/src/locales/en.json]: just `Please enter the correct password. `

current password label is displayed above the input

**Pull request merged**